src/internal/components/focus-lock/__integ__/focus-lock.test.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import useBrowser from '@cloudscape-design/browser-test-tools/use-browser';
+import { BasePageObject } from '@cloudscape-design/browser-test-tools/page-objects';
+import createWrapper, { ElementWrapper } from '../../../../../lib/components/test-utils/selectors';
+
+function setupTest(testFn: (page: BasePageObject, wrapper: ElementWrapper) => Promise<void>) {
+  return useBrowser(async browser => {
+    await browser.url('#/light/focus-lock');
+    const page = new BasePageObject(browser);
+    const wrapper = createWrapper();
+    await page.waitForVisible(wrapper.findButton().toSelector());
+    await testFn(page, wrapper);
+  });
+}
+
+describe('FocusLock', () => {
```

Comment:
Can this be tested in a unit test instead?

Reply:
You could probably craft something that kinda works. But when it comes to things like focus management, I have much more confidence in an integration test. I try to use them sparingly, but I think in this case it's the better path.